drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -206,15 +227,47 @@ static int siwx917_scan(const struct device *dev, struct wifi_scan_params *z_sca
 		return -EBUSY;
 	}
 
-	/* The enum values are same, no conversion needed */
-	sl_scan_config.type = z_scan_config->scan_type;
+	if (z_scan_config->scan_type == WIFI_SCAN_TYPE_ACTIVE) {
+		sl_scan_config.type = SL_WIFI_SCAN_TYPE_ACTIVE;
+		if (z_scan_config->dwell_time_active) {
+			ret = sl_si91x_configure_timeout(SL_SI91X_CHANNEL_ACTIVE_SCAN_TIMEOUT,
+								z_scan_config->dwell_time_active);
+			if (ret) {
+				LOG_ERR("Failed to configure active dwell timeout: %d", ret);
+				return -EINVAL;
+			}
+		}
+	} else {
+		sl_scan_config.type = SL_WIFI_SCAN_TYPE_PASSIVE;
+		if (z_scan_config->dwell_time_passive) {
+			ret = sl_si91x_configure_timeout(SL_SI91X_CHANNEL_PASSIVE_SCAN_TIMEOUT,
+								z_scan_config->dwell_time_passive);
+			if (ret) {
+				LOG_ERR("Failed to configure passive dwell timeout: %d", ret);
+				return -EINVAL;
+			}
+		}
+	}
 
-	sl_scan_config.channel_bitmap_2g4 = 0xFFFF;
-	memset(sl_scan_config.channel_bitmap_5g, 0xFF, sizeof(sl_scan_config.channel_bitmap_5g));
+#ifdef CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
```

Comment:
`WIFI_MGMT_SCAN_CHAN_MAX_MANUAL` is always defined. You don't need this `#ifdef`